core/HoistBaseDecorators.js

```diff
@@ -0,0 +1,93 @@
+/*
```

Comment:
Should/could we validate (or at least comment) that these decorators can _only_ be used on classes which extend `HoistBase`?  The filename makes that pretty clear, but might be worth stressing in comments and/or having these throw.

Reply:
Great idea -- did this, and actually caught a case where we were using the decorator and it was doing nothing!  

👍